Trace View docs change (#21250)
DataDog/documentation: content/en/tracing/trace_explorer/trace_view.md

  text: "Understand how to read a Datadog Trace"
---

{{< img src="tracing/trace_view/trace_view.png" alt="Trace View" style="width:90%;">}}

## Overview

View an individual [trace][1] to see all of its [spans][2] and associated metadata. Each trace can be visualized as either a Flame Graph, Span List, Waterfall, or Map.

The trace header displays critical trace information, including the root span's service name, resource name, trace ID, end-to-end trace duration, and the trace start datetime. To retain a trace indefinitely, click **Open Full Page** and save the URL.

{{< img src="tracing/trace_view/trace_header.png" alt="Trace header" style="width:90%;">}}

## Trace visualizations

{{< tabs >}}
{{% tab "Flame Graph" %}}

{{< img src="tracing/trace_view/flamegraph.png" alt="Flame Graph" style="width:90%;">}}

> The default visualization that displays all the spans for an associated trace, color-coded on a timeline. This is useful for understanding the execution path of a request and where time was spent over a trace.

Reviewer: ```suggestion
Flame Graph is the default visualization that displays all the color-coded spans from a trace on a timeline. This is useful for understanding the execution path of a request and where time was spent over a trace.
```

To navigate the graph, scroll to zoom, click and drag to move around, and use the minimap to zoom into the selected span or zoom out to the full trace.

> The legend details the color coding of the flame graph. Group spans by either **Service** (default), **Host**, or **Container**. Choose to display either the percentage of trace execution time (**% Exec Time**) or span count (**Spans**) by color grouping. If errors exist on spans in the trace, highlight them in the flame graph by selecting the **Errors** checkbox under **Filter Spans**.

Reviewer: ```suggestion
The legend details the color coding of the flame graph. Group spans by either **Service** (default), **Host**, or **Container**. Choose to display either the percentage of trace execution time (**% Exec Time**) or span count (**Spans**) by group. If errors exist on spans in the trace, highlight them in the flame graph by selecting the **Errors** checkbox under **Filter Spans**.
```

{{< img src="tracing/trace_view/flamegraph_legend.mp4" alt="Flame Graph legend" video="true" style="width:90%;">}}

{{% /tab %}}
{{% tab "Span List" %}}

{{< img src="tracing/trace_view/spanlist.png" alt="Trace View" style="width:90%;">}}

> Displays [resources][1] by grouping ([service][2] by default) and sorts them according to their count of spans. This visualization is useful for scanning latency information by resource or grouping.

Reviewer: ```suggestion
Displays [resources][1] by group ([service][2] by default) and sorts them according to their count of spans. This visualization is useful for scanning latency information by resource or grouping.
```

Filter resources by type or naming information using the corresponding buttons and text-based search.

{{< img src="tracing/trace_view/spanlist_headers.png" alt="Span List headers" style="width:90%;">}}

> Sort groupings by clicking on the corresponding column header: **RESOURCE**, **SPANS**, average duration (**AVG DURATION**), execution time (**EXEC TIME**), or percentage of trace execution time (**% EXEC TIME**).

Reviewer: ```suggestion
Groups can be sorted by clicking on the corresponding column header: **RESOURCE**, **SPANS**, average duration (**AVG DURATION**), execution time (**EXEC TIME**), or percentage of trace execution time (**% EXEC TIME**).
```

[1]: /tracing/glossary/#resources
[2]: /tracing/glossary/#services
{{% /tab %}}
{{% tab "Waterfall (Beta)" %}}

<div class="alert alert-info"><strong>Join the Beta!</strong><br />
To join the Waterfall private beta <a href="https://forms.gle/LjJR1ZbF1tNDv5JC6">fill out this form</a>.</div>

{{< img src="tracing/trace_view/waterfall.png" alt="Waterfall" style="width:90%;">}}

> Displays all the spans for an associated trace, color-coded on separate rows and on a timeline. This visualization is useful for isolating relevant parts of a trace and understanding how asynchronous processes fit into the execution flow.

Reviewer: ```suggestion
Displays all the spans for an associated trace, color-coded on separate rows and on a timeline. This visualization is useful for isolating and focusing relevant parts of a trace.
```

On each row (that is, per span):
* A bar (colored by service), whose length corresponds to the percentage of total trace duration

> * The service name, operation name, and resource name, with font styling: **service** operation <span style="color:gray">resource</span>

Reviewer: This sentence is not clear to me

Author: It refers to the visual part of the visualization, where each span has a bar:
![image](https://github.com/DataDog/documentation/assets/88278697/0595bda1-5abc-4dcd-9564-ed5b35fe42be)
Any recommendations for how to word this differently? It's tough to describe and easy to see/understand

Reviewer: The first bullet point on the bar is fine to me. it is more regarding the service name/operation/resource.
What about annotating on a screenshot?

* Absolute and relative span duration information
* (when applicable) An error icon or HTTP status code

To expand or collapse span descendants, click the plus or minus button on any row. To expand all spans, click the plus button to the left of the timescale.

{{% /tab %}}
{{% tab "Map" %}}

{{< img src="tracing/trace_view/map.png" alt="Map" style="width:90%;">}}

> Displays a service-level representation of the trace, with arrows showing the flow of calls between services. This visualization is useful for getting a high-level overview of the trace and the order in which services are called.

Reviewer: ```suggestion
Displays a representation of the trace of all the services involved in the trace. This visualization is useful for getting a high-level overview of the services dependencies and transaction lifecycle at a service level.
```

Hover over a service to highlight its parent and children, and click on it to focus on the service entry span.

{{% /tab %}}
{{< /tabs >}}

## More information

The height-adjustable bottom of the Trace View shows selected span and trace information. 

The span header contains service, operation, and resource names of the selected span as well as latency information. Pivot to other parts of the platform or narrow down your [Trace Explorer][5] search by clicking on the naming pill.

{{< img src="tracing/trace_view/span_header.png" alt="Span header" style="width:90%;">}}

> {{< tabs >}}

Reviewer: The screenshots of this section are all super old. @lucasdmaley can you update them and do a quick check on the text at the same time?

Author: Resolved with most recent commit

  text: "Understand how to read a Datadog Trace"
---

> {{< img src="tracing/trace_view/trace_view.png" alt="Trace View" style="width:90%;">}}

Reviewer 2: Minor comment, but I think we could potentially crop out the Datadog left-navigation and/or the page behind the actual trace view. I think that would help make this screenshot less overwhelming.

## Overview

View an individual [trace][1] to see all of its [spans][2] and associated metadata. Each trace can be visualized as either a Flame Graph, Span List, Waterfall, or Map.

> The trace header displays critical trace information, including the root span's service name, resource name, trace ID, end-to-end trace duration, and the trace start datetime. To retain a trace indefinitely, click \"Open Full Page\" and save the URL.

Reviewer 2: ```suggestion
The trace header displays critical trace information, including the root span's service name, resource name, trace ID, end-to-end trace duration, and the trace start datetime. To retain a trace indefinitely, click **Open Full Page** and save the URL.
```
What do we mean by "retain a trace indefinitely"? I think this means if you want to quickly revisit a trace, save the URL somewhere. It immediately made me think of [trace retention](https://docs.datadoghq.com/tracing/trace_pipeline/trace_retention/), so I think we should use a different word if these aren't related concepts.

Author: These actually are related concepts - by clicking on **Open Full Page**, that specific trace is retained indefinitely instead of the default retention period (and can be accessed by the user at any time later via the URL)

{{< img src="tracing/trace_view/trace_header.png" alt="Trace header" style="width:90%;">}}

> Selected span information is visible at the bottom of the Trace View.

Reviewer 2: ```suggestion
```
I'm not sure if this adds any valuable info. Might be best to just remove it.

## Trace visualizations

{{< tabs >}}
{{% tab "Flame Graph" %}}

{{< img src="tracing/trace_view/flamegraph.png" alt="Flame Graph" style="width:90%;">}}

> The default visualization. Displays all the spans for an associated trace, color-coded on a timeline. Particularly useful for understanding the execution path of a request and where time was spent over a trace.

Reviewer 2: ```suggestion
The default visualization that displays all the spans for an associated trace, color-coded on a timeline. This is useful for understanding the execution path of a request and where time was spent over a trace.
```

To navigate the graph, scroll to zoom, click and drag to move around, and use the minimap to zoom into the selected span or zoom out to the full trace.

> The legend details the color coding of the flame graph. Group spans by either **service** (default), **host**, or **container** - and choose to display either **percentage of trace execution time** or **span count** by color grouping. If errors exist on spans in the trace, highlight them in the flame graph by ticking the Errors checkbox under Filter Spans.

Reviewer 2: ```suggestion
The legend details the color coding of the flame graph. Group spans by either **Service** (default), **Host**, or **Container**. Choose to display either the percentage of trace execution time (**% Exec Time**) or span count (**Spans**) by color grouping. If errors exist on spans in the trace, highlight them in the flame graph by selecting the **Errors** checkbox under **Filter Spans**.
```
My main suggestions here are aligning the capitalization/wording with the UI.

{{< img src="tracing/trace_view/flamegraph_legend.mp4" alt="Flame Graph legend" video="true" style="width:90%;">}}

{{% /tab %}}
{{% tab "Span List" %}}

{{< img src="tracing/trace_view/spanlist.png" alt="Trace View" style="width:90%;">}}

> Displays [resources][1] by grouping ([service][2] by default) and sorts them according to their count of spans. Particularly useful for scanning latency information by resource or grouping.

Reviewer 2: ```suggestion
Displays [resources][1] by grouping ([service][2] by default) and sorts them according to their count of spans. This visualization is useful for scanning latency information by resource or grouping.
```

Filter resources by type or naming information using the corresponding buttons and text-based search.

{{< img src="tracing/trace_view/spanlist_headers.png" alt="Span List headers" style="width:90%;">}}

> Sort groupings by clicking on the corresponding column header: **resource**, **spans**, **average duration**, **execution time**, or **percentage of trace execution time**.

Reviewer 2: ```suggestion
Click a column heading to sort by that information.
```
I personally don't think we need to explicitly list all of the columns.

Author: Did so to avoid conflation with the buttons which exist immediately above them
![image](https://github.com/DataDog/documentation/assets/88278697/b1081ed9-7a7c-426c-93aa-c4812604b8db)
If you're not worried about this, your suggestion is fine by me. If you think the extra clarification is helpful, we could either:
- Have the list of columns (capitalization/grammar matching the UI)
or
- Have a screenshot of the column headers
Up to you!

Reviewer 2: Fair enough! That's a valid point. Let's keep what you have but match the text to how it appears in the UI.

[1]: /tracing/glossary/#resources
[2]: /tracing/glossary/#services
{{% /tab %}}
{{% tab "Waterfall (Beta)" %}}

<div class="alert alert-info"><strong>Join the Beta!</strong><br />
To join the Waterfall private beta <a href="https://forms.gle/LjJR1ZbF1tNDv5JC6">fill out this form</a>.</div>

{{< img src="tracing/trace_view/waterfall.png" alt="Waterfall" style="width:90%;">}}

> Displays all the spans for an associated trace, color-coded on separate rows and on a timeline. Particularly useful for isolating relevant parts of a trace and understanding how asynchronous processes fit into the execution flow.

Reviewer 2: ```suggestion
Displays all the spans for an associated trace, color-coded on separate rows and on a timeline. This visualization is useful for isolating relevant parts of a trace and understanding how asynchronous processes fit into the execution flow.
```

On each row (that is, per span):
* A bar (colored by service), whose length corresponds to the percentage of total trace duration
* The service name, operation name, and resource name, with font styling: **service** operation <span style="color:gray">resource</span>
* Absolute and relative span duration information
* (when applicable) An error icon or HTTP status code

> Expand/collapse span descendants by clicking the plus/minus button on each row, and expand all spans with the plus button to the left of the timescale.

Reviewer 2: ```suggestion
To expand or collapse span descendants, click the plus or minus button on any row. To expand all spans, click the plus button to the left of the timescale.
```

{{% /tab %}}
{{% tab "Map" %}}

{{< img src="tracing/trace_view/map.png" alt="Map" style="width:90%;">}}

> Displays a service-level representation of the trace, with arrows showing the flow of calls between services. Particularly useful for getting a high level overview of the trace and the order in which services are called.

Reviewer 2: ```suggestion
Displays a service-level representation of the trace, with arrows showing the flow of calls between services. This visualization is useful for getting a high-level overview of the trace and the order in which services are called.
```
